scripts/release/select_release_assets.py

```
@@ -0,0 +1,154 @@
+        _ensure_file(repo_root / metadata_path)
+
+    pattern = _artifact_pattern(effective_sha)
+    installers_by_target: dict[tuple[str, str], set[str]] = {
+        target: set() for target in TARGET_INSTALLER_SUFFIXES
+    }
+    sidecars_by_installer: dict[str, set[str]] = {}
+    selected_artifacts: list[str] = []
+
+    for artifact_path in sorted(artifacts_dir.iterdir(), key=lambda path: path.name):
+        if not artifact_path.is_file():
+            raise ValueError(f"unexpected release artifact path: {artifact_path.name}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)